qiskit/circuit/library/standard_gates/sx.py

```diff
@@ -0,0 +1,258 @@
+
+    .. note::
+
+        A global phase difference exists between the definitions of
+        :math:`RX(\pi/2)` and :math:`\sqrt{X}`.
+
+        .. math::
+
+            RX(\pi/2) = \frac{1}{\sqrt{2}} \begin{pmatrix}
+                        1 & -i \\
+                        -i & 1
+                      \end{pmatrix}
+                    = e^{-i pi/4} \sqrt{X}
+
+    """
+
+    def __init__(self, label='√X'):
+        """Create new SX gate."""
+        super().__init__('sx', 1, [], label=label)
+
+    def _define(self):
+        """
+        gate sx a { rz(-pi/2) a; h a; rz(-pi/2); }
+        """
+        # pylint: disable=cyclic-import
+        from qiskit.circuit.quantumcircuit import QuantumCircuit
+        from .rz import RZGate
+        from .h import HGate
+        q = QuantumRegister(1, 'q')
+        qc = QuantumCircuit(q, name=self.name)
+        rules = [
+            (RZGate(-pi / 2), [q[0]], []),
+            (HGate(), [q[0]], []),
+            (RZGate(-pi / 2), [q[0]], [])
+        ]
+        qc.data = rules
+        self.definition = qc
+
+    def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None):
+        """Return a (multi-)controlled-SX gate.
+
+        One control returns a CSX gate.
+
+        Args:
+            num_ctrl_qubits (int): number of control qubits.
+            label (str or None): An optional label for the gate [Default: None]
+            ctrl_state (int or str or None): control state expressed as integer,
+                string (e.g. '110'), or None. If None, use all 1s.
+
+        Returns:
+            ControlledGate: controlled version of this gate.
+        """
+        if num_ctrl_qubits == 1:
+            gate = CSXGate(label=label, ctrl_state=ctrl_state)
+            gate.base_gate.label = self.label
+            return gate
+        return super().control(num_ctrl_qubits=num_ctrl_qubits, label=label, ctrl_state=ctrl_state)
+
+    # Differs by global phase of exp(-i pi/4) with correct RZ.
+    # If the RZ == U1, then the global phase difference is exp(i pi/4)
+    # TODO: Restore after allowing phase on circuits.
+    # def to_matrix(self):
+    #     """Return a numpy.array for the SX gate."""
+    #     return numpy.array([[1 + 1j, 1 - 1j],
+    #                         [1 - 1j, 1 + 1j]], dtype=complex) / 2
+
+
+class SXdgGate(Gate):
+    r"""The inverse single-qubit Sqrt(X) gate.
+
+    .. math::
+
+        \sqrt{X}^{\dagger} = \frac{1}{2} \begin{pmatrix}
+                1 - i & 1 + i \\
+                1 + i & 1 - i
+            \end{pmatrix}
+
+
+    .. note::
+
+        A global phase difference exists between the definitions of
+        :math:`RX(-\pi/2)` and :math:`\sqrt{X}^{\dagger}`.
+
+        .. math::
+
+            RX(-\pi/2) = \frac{1}{\sqrt{2}} \begin{pmatrix}
+                        1 & i \\
+                        i & 1
+                      \end{pmatrix}
+                    = e^{-i pi/4} \sqrt{X}^{\dagger}
+
+    """
+
+    def __init__(self, label='√X_dg'):
```

Comment:
This would be similar to how sdg and tdg are rendered. 
```suggestion
    def __init__(self, label='√X^†'):
```

Reply:
Also I noticed this will draw as two columns in the mpl drawer, but you can fix that by adding it to the list of standard gates on line 704 of visualization/matplotlib.py

Reply:
also it would be good to have some consistency between this and other "daggered" gates in the text drawer. 
because Sdg and Tdg don't have labels set, like this one, they draw with their name so they look like:

<img width="141" alt="image" src="https://user-images.githubusercontent.com/8622381/88447441-453aec00-ce01-11ea-99cc-69813e7ced33.png">

Also since text-drawer doesn't have math mode the exponent doesn't show well. What's the best way to fix this?

I'm being nit-picky on these because they will end up being shown very prominently when we translate to these basis.

Reply:
Not sure about the text drawer, maybe it is really `S^†`, alternatively `S†` or -- looking at the list of ascii superscript characters, `S⁺` (small plus)?

Reply:
Turns out that the dagger sign causes a test to fail, which says that only [Code page 437](https://en.wikipedia.org/wiki/Code_page_437) symbols are allowed. @1ucian0 is that because other symbols might not get rendered on different machines or what's the reason for this?

Reply:
Looking at the available symbols from this page I switched back to `_dg` for now.

Reply:
@1ucian0 why can't we support symbols outside of that list?